src/backend/distributed/commands/statistics.c

```diff
@@ -506,6 +518,65 @@ GetExplicitStatisticsSchemaIdList(Oid relationId)
 }
 
 
+/*
+ * GetAlterIndexStatisticsCommands returns the list of ALTER INDEX .. ALTER COLUMN ..
+ * SET STATISTICS commands, based on non default targets of the index with given id.
+ * Note that this function only looks for expression indexes, since this commands is
+ * valid for only expression indexes.
+ */
+static List *
+GetAlterIndexStatisticsCommands(Oid indexOid)
+{
+	List *alterIndexStatisticsCommandList = NIL;
+	int exprCount = 0;
+	while (true)
+	{
+		StringInfoData attName;
+		initStringInfo(&attName);
+
+		/* statistics can be set for an index only if it's an expressional index */
+		appendStringInfoString(&attName, "expr");
+
+		if (exprCount > 0)
+		{
+			appendStringInfo(&attName, "%d", exprCount);
+		}
+
+		HeapTuple tuple = SearchSysCacheAttName(indexOid, attName.data);
+
+		if (!HeapTupleIsValid(tuple))
+		{
+			break;
+		}
+
+		Form_pg_attribute targetAttr = (Form_pg_attribute) GETSTRUCT(tuple);
+		if (targetAttr->attstattarget != -1)
+		{
+			char *indexName = get_rel_name(indexOid);
+			Oid schemaOid = get_rel_namespace(indexOid);
+			char *schemaName = get_namespace_name(schemaOid);
+
+			StringInfoData command;
+			initStringInfo(&command);
+			appendStringInfo(&command,
+							 "ALTER INDEX %s ALTER COLUMN %d SET STATISTICS %d",
+							 quote_qualified_identifier(schemaName, indexName),
```

Comment:
I wonder could  we directly use generate_qualified_relation_name instead of quote_qualified_identifier